### PR TITLE
Phase2-hgx275 Third step of making the V15 geometry

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Third step of making the V15 geometry (algorithms of making full silicon layers)

#### PR validation:

Tested by dumping the geometry and making a check with cmsShow

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cvuosalo @kpedro88 @srimanob Please review and approve this PR

**Comment by cvuosalo** [comment]
@bsunanda I notice that false file names are being used inside the XML files; e.g., `<PosPartSection label="hgcalwaferpart.xml" eval="true">` inside of `hgcalwafer.xml`. Won't that practice possibly cause confusion?

File: Geometry/HGCalCommonData/plugins/DDHGCalSiliconModule.cc [hunk 1]
```diff
@@ -0,0 +1,449 @@
+///////////////////////////////////////////////////////////////////////////////
+// File: DDHGCalSiliconModule.cc
+// Description: Geometry factory class for HGCal (EE and HESil) using
+//              information from the file
+///////////////////////////////////////////////////////////////////////////////
+
+#include "DataFormats/Math/interface/GeantUnits.h"
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/plugins/DDHGCalSiliconModule.cc:7]: It looks like this file uses only angle utilities, not units utilities. If so, it is better to change the include:
```
#include "DataFormats/Math/interface/angle_units.h"
```
Don't bring more dependencies than necessary into the file.

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalSiliconModule.cc [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+#include <vector>
+
+#include "Geometry/HGCalCommonData/interface/HGCalGeomTools.h"
+#include "Geometry/HGCalCommonData/interface/HGCalParameters.h"
+#include "Geometry/HGCalCommonData/interface/HGCalProperty.h"
+#include "Geometry/HGCalCommonData/interface/HGCalTypes.h"
+#include "Geometry/HGCalCommonData/interface/HGCalWaferIndex.h"
+#include "Geometry/HGCalCommonData/interface/HGCalWaferType.h"
+#include "DD4hep/DetFactoryHelper.h"
+#include "DataFormats/Math/interface/CMSUnits.h"
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalSiliconModule.cc:20]: Please change the include to `angle_units.h`. DD4hep algorithms cannot be tied to CMS units anymore.

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalSiliconModule.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalSiliconModule.cc]: Done

**Comment by bsunanda** [comment]
@cvuosalo  @srimanob @kpedro88 Please review this PR and approve this

**Comment by bsunanda** [comment]
@cvuosalo @srimanob @kpedro88 Please approve this PR

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**